File: src/main/java/com/lifecycle/components/entities/Field.java
package com.lifecycle.components.entities;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Field {
	private String name;
	private String description;
	private String type;
	private String uom;
	private String scalar;
	private Long decimals;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public String getUom() {
		return uom;
	}

	public void setUom(String uom) {
		this.uom = uom;
	}

	public String getScalar() {
		return scalar;
	}

	public void setScalar(String scalar) {
		this.scalar = scalar;
	}

	public Long getDecimals() {
		return decimals;
	}

	public void setDecimals(Long decimals) {
		this.decimals = decimals;
	}

}

File: src/main/resources/4thYrProjectWebsite/app-framework/data_structures/metadata/field.js
'use strict';

import JsonObject from '../../base/json-object.js';

/**
 * Metadata field element
 * @module metadata/field
 * @extends JsonObject
 */
export default class Field extends JsonObject { 
	
	get name() { return this.json["name"]; }
	
	get description() { return this.json["description"]; }
	
	get type() { return this.json["type"]; }
	
	get uom() { return this.json["uom"]; }
	
	get scalar() { return this.json["scalar"]; }
	/*
	static make(name) {
		return new Field({
			"name": name
		});
	}
	*/
	static make(name) {
		return {
			"name": name,
			"description": null,
			"type": null,
			"uom": null,
			"scalar": null
		};
	}
}